#2152 Can't activate more than 1 Bacnet IP datasource simultaneously

Patrykb0802 opened the pull request:
Added:
- Port validation for BACnet data sources (checks if its taken or not)
- Additional info about ports in help description in BACnet data source edit view

Patrykb0802 committed:
#2152 Can't activate more than 1 Bacnet IP datasource simultaneously

fix/#2152_Cant_activate_more_than_1_Bacnet_IP_datasource_simultaneously

Patrykb0802 committed:
#2152 Can't activate more than 1 Bacnet IP datasource simultaneously

Patrykb0802 committed:
#2152 Can't activate more than 1 Bacnet IP datasource simultaneously

Patrykb0802 committed:
#2152 Can't activate more than 1 Bacnet IP datasource simultaneously

Limraj commented on src/org/scada_lts/dao/DAO.java:
```diff
@@ -169,5 +173,23 @@ public boolean isTest() {
 		this.test = test;
 	}
+
+	public boolean isBacNetDataSourcePortUsed(int excludeId, int port) {
```
Instead of doing this in the database, a more efficient approach to checking if a given port is currently in use would be to search the running Data Sources in the RuntimeManager. 
There should also be no such method in dao, this method should be in only Service.

We always try to use what is already there, and do not create our own solutions if the solution exists, e.g. here a better approach would be to use the method from Service:
`List<DataSourceVO<?>> getDataSources(DataSourceVO.Type type)`

Limraj commented on webapp-resources/messages_de.properties:
```diff
@@ -3365,4 +3365,5 @@ header.imageSets=Image sets
-systemsettings.reports.dataPointExtendedNameLengthLimit=Point name length limit in reports
\ No newline at end of file
+systemsettings.reports.dataPointExtendedNameLengthLimit=Point name length limit in reports
+Bacnet.invalid.port=Port already in use: cannot bind
```
bacnet with lower case letter

Limraj reviewed (commented):
I suggest:

1. Check the ports used in the running Data Source from RuntimeManger;

2. **Add a check if a given port is actually not used in the system, by trying to connect;** Use implementation from:
![image](https://github.com/user-attachments/assets/50321ded-e0d4-4397-9ab3-770cdbf74202)
This is the way to check whether a given port is free.

3. At the very end, you can add a check of what is saved in the database, but moving the implementation to Service and using existing methods from service in it. Unfortunately, we will not achieve anything by moving the method to dao, because this information is stored in a column with a blob, so direct queries and deserialization do not speed up this process.

4. It is also worth checking what event is generated in Data Source Bacnet if a given port is used, whether the content of this event is readable to the user.

Patrykb0802 committed:
#2152 Can't activate more than 1 Bacnet IP datasource simultaneously

Patrykb0802 committed:
#2152 Can't activate more than 1 Bacnet IP datasource simultaneously

Limraj committed:
Merge branch 'release/2.8.0' into fix/#2152_Cant_activate_more_than_1_Bacnet_IP_datasource_simultaneously

Limraj commented:
After our conversation, I suggest you focus on the following points:
2. Add a check if a given port is actually not used in the system, by trying to connect; 
4. It is also worth checking what event is generated in Data Source Bacnet if a given port is used, whether the content of this event is readable to the user.

Limraj committed:
Merge branch 'release/2.8.0' into fix/#2152_Cant_activate_more_than_1_Bacnet_IP_datasource_simultaneously

Limraj committed:
Merge remote-tracking branch 'origin/release/2.8.0' into fix/#2152_Cant_activate_more_than_1_Bacnet_IP_datasource_simultaneously

# Conflicts:
#	webapp-resources/messages_es.properties
#	webapp-resources/messages_fi.properties
#	webapp-resources/messages_fr.properties
#	webapp-resources/messages_lu.properties
#	webapp-resources/messages_nl.properties
#	webapp-resources/messages_pl.properties
#	webapp-resources/messages_pt.properties
#	webapp-resources/messages_ru.properties
#	webapp-resources/messages_zh.properties

Limraj committed:
#2152 Can't activate more than 1 Bacnet IP datasource simultaneously:
- Created BACnetUtils;
- Moved method from DataSourceUtils to BACnetUtils;
- Moved invoke checkFreePort from BACnetIPDataSourceVO to DataSourceEditDwr;

Limraj committed:
#2152 Can't activate more than 1 Bacnet IP datasource simultaneously:
- Corrected file helps

Limraj committed:
#2152 Can't activate more than 1 Bacnet IP datasource simultaneously:
- Update bacnetIpDS.htm

Limraj committed:
#2152 Can't activate more than 1 Bacnet IP datasource simultaneously:
- set utf-8 encoding file in FileReader;
